### PR TITLE
Add initial version of in-source documentation

### PR DESCRIPTION
**Pull request by egavrin**
JerryScript-DCO-1.0-Signed-off-by: Evgeny Gavrin e.gavrin@samsung.com

If you have any suggestions please feel free to propose enhancements.

### CODE REVIEW

**Comment by seanshpark** [comment]
+1

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+To build debug version for Linux without LTO (Link Time Optimization):
+```bash
+LTO=OFF make debug.linux -j
+```
+
+### Checking Patch
+```bash
+make precommit -j
+```
+Sometimes pre-commit testing fails, in that case you'll see message like that:
```
Comment (ruben-ayrapetyan) [inline docs/DEVELOPMENT.md]: `If some style guidelines, build or test runs fail during precommit, then this is indicated with a message like this:`

**Comment by ruben-ayrapetyan** [comment]
Looks good to me

File: docs/DEVELOPMENT.md (hunk 1 above)
Comment (egavrin) [inline docs/DEVELOPMENT.md]: Fixed

**Comment by egavrin** [comment]
@zherczeg @galpeter could you please check this?

**Comment by egavrin** [comment]
@galpeter @zherczeg @LaszloLango may I merge this?

**Comment by LaszloLango** [comment]
I have no objection to merge this. +1 lgtm

**Comment by egavrin** [comment]
Thank you!

**Pull request merged**